From: Gabriel Valcazar <gabriel.valcazar@digi.com>
Date: Fri, 19 Jul 2019 10:42:35 +0200
Subject: [PATCH] iMX8QX: remove SC_BD_FLAGS_ALT_CONFIG flag in
 flash_regression_linux_m4 target

This flag breaks the ccimx8x U-Boot due to the code that is executed in the
SCFW when it's set. Make the build command the same as the "flash_all" target
in the rel_imx_4.14.78_1.0.0_ga tag, since the M4 demos still seem to work as
expected when the flag is removed. The "-p3" parameter also needs to be removed
for the images to boot properly.

https://jira.digi.com/browse/DEL-6677

Signed-off-by: Gabriel Valcazar <gabriel.valcazar@digi.com>
---
 iMX8QX/soc.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iMX8QX/soc.mak b/iMX8QX/soc.mak
index e08cf3c..88f990a 100755
--- a/iMX8QX/soc.mak
+++ b/iMX8QX/soc.mak
@@ -120,7 +120,7 @@ flash_linux_m4_xip: $(MKIMG) $(AHAB_IMG) scfw_tcm.bin u-boot-atf-container.img m
 	./$(QSPI_PACKER) $(QSPI_HEADER)
 
 flash_regression_linux_m4: $(MKIMG) $(AHAB_IMG) scfw_tcm.bin u-boot-atf.bin m4_image.bin
-	./$(MKIMG) -soc QX -rev B0 -append $(AHAB_IMG) -c -flags 0x00200000 -scfw scfw_tcm.bin -ap u-boot-atf.bin a35 0x80000000 -p3 -m4 m4_image.bin 0 0x34FE0000 -out flash.bin
+	./$(MKIMG) -soc QX -rev B0 -append $(AHAB_IMG) -c -scfw scfw_tcm.bin -ap u-boot-atf.bin a35 0x80000000 -m4 m4_image.bin 0 0x34FE0000 -out flash.bin
 
 flash_regression_linux_m4_ddr: $(MKIMG) $(AHAB_IMG) scfw_tcm.bin u-boot-atf.bin m4_image.bin
 	./$(MKIMG) -soc QX -rev B0 -append $(AHAB_IMG) -c -flags 0x00200000 -scfw scfw_tcm.bin -ap u-boot-atf.bin a35 0x80000000 -p3 -m4 m4_image.bin 0 0x88000000 -out flash.bin
